soc/xtensa/intel_adsp/common/include/cpu_init.h

```diff
@@ -16,52 +17,7 @@ static ALWAYS_INLINE void cpu_early_init(void)
 {
 	uint32_t reg;
 
-#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V25
```

Comment:
Why not just change this to be an SoC feature check via a Kconfig  instead of checking for specific SoCs and enable this for SoCs supporting this instead? 
This will avoid duplication of code and will keep the SoC initialization path in one place.
Also note that this is standard code flow for xtensa, see `arch/xtensa/core/startup/reset-vector.S` for example.

Reply:
I thought about it, that is indeed a solution. Was just afraid that upcoming targets could have different initialization.
But, lets deal with the closes alligator. Will change it to use a Kconfig option.